lib/cli/watch-run.js

```diff
@@ -348,15 +360,26 @@ const eraseLine = () => {
 /**
  * Blast all of the watched files out of `require.cache`
  * @param {FSWatcher} watcher - chokidar FSWatcher
+ * @param {boolean} blastAll
  * @ignore
  * @private
  */
-const blastCache = watcher => {
-  const files = getWatchedFiles(watcher);
-  files.forEach(file => {
-    delete require.cache[file];
-  });
-  debug('deleted %d file(s) from the require cache', files.length);
+const blastCache = (watcher, blastAll) => {
+  if (blastAll) {
+    Object.keys(require.cache)
+      // Avoid deleting mocha binary (at minimum, breaks Mocha's watch tests)
+      .filter(file => !file.includes('/mocha/bin/'))
+      .forEach(file => {
+        delete require.cache[file];
+      });
+    debug('deleted all files from the require cache');
+  } else {
+    const files = getWatchedFiles(watcher);
+    files.forEach(file => {
+      delete require.cache[file];
+    });
+    debug('deleted %d file(s) from the require cache', files.length);
+  }
```

Comment:
[Style] Totally optional, but we could deduplicate the internals of this function a bit:

```js
  const files = blastAll
    ? Object.keys(require.cache)
      // Avoid deleting mocha binary (at minimum, breaks Mocha's watch tests)
      .filter(file => !file.includes('/mocha/bin/'))
    : getWatchedFiles(watcher);

  files.forEach(file => {
    delete require.cache[file];
  });

  debug('deleted %d file(s) from the require cache (blastAll: %s)', files.length, blastAll);
```

No worries if you hate this 🙂

Reply:
If we always clean the whole cache, then there's no need at all for the conditional logic, so I removed it. What I did like and left was the count of deleted files for the `debug` statement, might be useful.